chore(agw): build python precommit container locally

wolfseb opened the pull request:
Part of #15000 

## Summary

Build the python precommit container locally instead of downloading the image. The build takes <1min, and the CI respective jobs also build the image locally, so it should not be necessary to upload the container image.

## Test Plan

Run the script.

## Additional Information

- [ ] This change is backwards-breaking

Commit:
chore(agw): build python precommit container locally

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

Commit:
fix(agw): wemake-python linter changes

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

voisey reviewed (commented):
Could you please remove
```
# build the base image
./precommit.py --build
```
from https://github.com/magma/magma/blob/master/docs/readmes/lte/dev_unit_testing.md#format-python as this will longer work?

Commit:
docs: remove build command for format python

The build is now always run (the cache is used on subsequent runs),
as the container will not be downloaded anymore.

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

LKreutzer reviewed (commented):
Does it actually make sense to publish the image in CI then? It looks like the publishing could just be stopped.

Commit:
chore(ci): do not push python precommit image to registry

Change the behaviour of the AGW Build Docker Image Python Precommit workflow so it only checks if the image can be built, but do not push it to the registry, since the pushed image is not used anywhere anymore.

Signed-off-by: Sebastian Wolf <sebastian.wolf@tngtech.com>

voisey commented on .github/workflows/docker-builder-python-precommit.yml:
```diff
@@ -51,4 +51,3 @@ jobs:
           IMAGE_STREAM: ${{ env.IMAGE_STREAM }}
           IMAGE_TAGS: ${{ env.IMAGE_TAGS }}
           DOCKERFILE: ${{ env.DOCKERFILE }}
-          PUSH_TO_REGISTRY: ${{ github.event_name == 'push' && github.ref_name == 'master' && github.token != null }}
```
It seems this parameter is required, see [here](https://github.com/magma/magma/blob/master/.github/workflows/composite/docker-builder/action.yml#L31), so I guess it should just be set to false.

wolfseb replied:
Thanks, done!
